python/lib/dcos/dcos/mesos.py

```diff
@@ -1271,7 +1271,8 @@ def _run(self):
         try:
             if self.interactive:
                 tty.setraw(fd, when=termios.TCSANOW)
-                self._window_resize(signal.SIGWINCH, None)
+                self._window_resize(signal.SIGWINCH, dimensions=[0, 0])
+                self._window_resize(signal.SIGWINCH)
```

Comment:
Regarding the limitation in your commit message (`you are not able to resize the screen
and have it react properly`), would it make sense to catch the SIGWINCH signal and then call `_window_resize`? Maybe as a possible future improvement.

Reply:
Calling this function sends a message to the TTY on the remote machine that a local resize has happened. When we first attach I want to send two messages: one setting the remote sides size to 0, followed by a second message setting it to the real size of the local terminal. Doing this causes the remote end to handle its SIGWINCH signal, thus redrawing the screen, and having that reflected locally.

Im not sure exactly what you are suggesting here, because there will be no local SIGWINCH signal to catch initially. This is just an artificial resize triggered on the remote end to force it to redraw its screen.

Also note, this is the same technique used by the docker CLI to force a refresh on attach.

Reply:
Calling this function sends a message to the TTY on the remote machine that a local resize has happened. When we first attach I want to send two messages: one setting the remote sides size to 0, followed by a second message setting it to the real size of the local terminal. Doing this causes the remote end to handle its SIGWINCH signal, thus redrawing the screen, and having that reflected locally.

Im not sure exactly what you are suggesting here, because there will be no local SIGWINCH signal to catch initially. This is just an artificial resize triggered on the remote end to force it to redraw its screen.

Also note, this is the same technique used by the docker CLI to force a refresh on attach.

Reply:
I should add a comment in line about this regardless a d link to the conversation about how/where it was added fir docker.

Reply:
Ah I see, I missed the line just below (`signal.signal(signal.SIGWINCH, self._window_resize)`), that's what I meant. Thx for the info.